lib/validation/validation.tsx

```diff
@@ -144,6 +146,17 @@ const isFieldResponseValid = (
 
       break;
     }
+    case FormElementTypes.formattedDate: {
+      if (validator.required && !value) {
+        return t("input-validation.required");
+      }
+
+      if (value && !isValidDate(JSON.parse(value as string) as DateObject)) {
```

Comment:
FYI - looks like the error message link doesn't link to the component properly might need a wrapping id

<img width="587" alt="Screenshot 2024-08-26 at 9 28 30 AM" src="https://github.com/user-attachments/assets/3236365e-3a28-408c-9c63-d23561ea5b8c">